lib/tls_test_kit/tls_version_test.rb

```diff
@@ -78,30 +87,41 @@ def version_required?(version)
         begin
           http.request_get(uri)
           if self.class.version_forbidden? version
-            add_message('error', "Server incorrectly allowed #{version_string} connection.")
+            message =
+              "#{url} allowed #{version_string} connection even though #{version_string} connections should be denied."
+            incorrectly_permitted_tls_versions << version_string
+            incorrectly_permitted_tls_versions_messages << message
+
+            add_message(self.class.incorrectly_permitted_tls_version_message_type, message)
           elsif self.class.version_required? version
-            add_message('info', "Server correctly allowed #{version_string} connection as required.")
+            add_message('info', "#{url} correctly allowed #{version_string} connection as required.")
             tls_support_verified = true
           else
-            add_message('info', "Server allowed #{version_string} connection.")
+            add_message('info', "#{url} allowed #{version_string} connection.")
             tls_support_verified = true
           end
         rescue StandardError => e
           if self.class.version_required? version
-            add_message('error', "Server incorrectly denied #{version_string} connection: #{e.message}")
+            add_message('error', "#{url} incorrectly denied #{version_string} connection: #{e.message}")
           elsif self.class.version_forbidden? version
-            add_message('info', "Server correctly denied #{version_string} connection as required.")
+            add_message('info', "#{url} correctly denied #{version_string} connection as required.")
           else
-            add_message('info', "Server denied #{version_string} connection.")
+            add_message('info', "#{url} denied #{version_string} connection.")
           end
         end
       end
 
+      output incorrectly_permitted_tls_versions_messages: incorrectly_permitted_tls_versions_messages.join("\n")
```

Comment:
Can we clean this up a bit?  The language in that box is pretty redundant.  Something like:



```suggestion
      output incorrectly_permitted_tls_versions_messages: "#{url} did not deny TLS connections for version(s) #{incorrectly_permmited_tls_versions.join(",").  The system may deny content from being sent over this connection, but this must be manually verified.
```